### PR TITLE
Support preemptible TPU using GCE

### PR DESCRIPTION
**Pull request by zpcore**
# Description

We are running out of v4 and v5p on zone us-central2 and us-east5. Capacity team suggests us use preemptible TPU resources.

This PR adds the preemptible config in the TPU resource creation API.

# Tests

Preemptible TPU [test](https://60bb71acc6784780b3bbdbae4ffa636f-dot-us-central1.composer.googleusercontent.com/dags/pytorchxla-torchbench/grid?dag_run_id=manual__2024-08-02T00%3A27%3A04.293890%2B00%3A00&tab=logs&task_id=torchbench_BERT_pytorch-v4-8.provision.create_queued_resource.wait_for_ready_queued_resource)

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run one-shot tests and provided workload links above if applicable. 
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

**Comment by zpcore** [comment]
We still can't claim the v5p resource even using the preemptible resources. v4 looks fine in the test. I will give up the v5p for now...

**Comment by zpcore** [comment]
Ping to see if anyone can help make a review. Thanks!

File: dags/pytorch_xla/configs/pytorchxla_torchbench_config.py [hunk 1]
```diff
@@ -77,7 +77,7 @@ class R2_4(enum.Enum):
     TORCH = "torch==2.4.0"
     TORCHVISION = "torchvision==0.19.0"
     TORCHAUDIO = "torchaudio==2.4.0"
-    TORCH_XLA_GPU_DOCKER = "us-central1-docker.pkg.dev/tpu-pytorch-releases/docker/xla:r2.4.0-rc2_3.10_cuda_12.1"
+    TORCH_XLA_GPU_DOCKER = "us-central1-docker.pkg.dev/tpu-pytorch-releases/docker/xla:r2.4.0-rc8_3.10_cuda_12.1"
```
Comment (will-cromar) [inline dags/pytorch_xla/configs/pytorchxla_torchbench_config.py:80]: Do you want to use the RC or the final release tag?

**Review by will-cromar — APPROVED** [review]

File: dags/pytorch_xla/configs/pytorchxla_torchbench_config.py (hunk 1 above)
Comment (zpcore) [reply dags/pytorch_xla/configs/pytorchxla_torchbench_config.py]: Updated

**Pull request merged**